runtime/drivers/druid/druid.go

```diff
@@ -162,6 +174,53 @@ func (d *driver) TertiarySourceConnectors(ctx context.Context, src map[string]an
 	return nil, fmt.Errorf("not implemented")
 }
 
+func (d driver) checkVersion(dsn string) error {
+	parsedURL, err := url.Parse(dsn)
+	if err != nil {
+		return err
+	}
+	parsedURL.Path = "/status"
+	statusURL := parsedURL.String()
+
+	req, err := http.NewRequest(http.MethodGet, statusURL, http.NoBody)
+	if err != nil {
+		return err
+	}
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("druid version check failed with status code: %d", resp.StatusCode)
+	}
+
+	var statusResponse map[string]any
+	if err := json.NewDecoder(resp.Body).Decode(&statusResponse); err != nil {
+		return fmt.Errorf("failed to decode Druid status response: %w", err)
+	}
+
+	// Check if druid version is greater than or equal to 28.0.0, because of SQL NULL compatibility
+	if version, ok := statusResponse["version"]; ok {
```

Comment:
You can use an inline struct to make the JSON library decode the field into a string automatically:

```suggestion
	var statusResponse struct {
		Version string `json:"version"`
	}
	if err := json.NewDecoder(resp.Body).Decode(&statusResponse); err != nil {
		return fmt.Errorf("failed to decode Druid status response: %w", err)
	}
```